### ISSUE TITLE
New Lead Maintainer Wanted

### ISSUE THREAD

**Issue by nebrelbug** [squirrellyjs/squirrelly#210]
Hey everyone, this is @nebrelbug -- the lead maintainer of Squirrelly and its companion project, [Eta](https://eta.js.org).

On October 28, 2020, I will leave on a 2-year service opportunity. During that time, I will be unable to work on any of my open-source projects.

Since my date of departure is drawing closer, I've decided it is time to begin **looking for a new lead maintainer**.

## Maintainer Requirements

- Proficiency in TypeScript
- Open-source experience. Preferably the maintainer/developer of an open-source JS/TS project with 50+ stars
- Willing to test changes & not break anything important :wink:
- Willing to follow the Project Goals (see below)

**Maintainer Jobs**

- Fix bugs
- Answer questions

A maintainer could also update documentation, write tutorials, write plugins, write integrations (ex. for Fastify, Eleventy, Koa) etc.

I've also updated Eta's file-handling logic, and it would be great to bring these changes back to Squirrelly

## Project Goals

- Follow SemVer versioning guidelines
  - This means Squirrelly's public API should remain backwards-compatible (at least within v8.x.x)
- Remain lightweight
  - Browser build should never exceed 5KB minzipped
- Remain stable
- No significant changes to parsing engine or logic
  - Squirrelly's parser has been optimized quite a bit for performance and reliability, and any significant changes run too high a risk of impacting either

## Fallback Plan

There's always the possibility I will be unable to find a new lead maintainer before my departure. This is one of the reasons I've accelerated development so much. It's also the reason Squirrelly and Eta are so extensively tested.

I feel confident that Squirrelly is stable enough it could continue to function, if needs be, without maintenance for several years. It has a stable API, is well tested, and has decent documentation. Many features can be added as 3rd-party plugins rather than core parts of the library. Moreover, Squirrelly's small size reduces the likelihood that important bugs exist inside the code.

That scenario, though, would be far from ideal, so I've made several backup plans:

- Give repository + npm access to a friend I trust to act in my behalf. This friend would not modify Squirrelly's code, but could give access if, while I was gone, someone asked to become a maintainer
- Add the project to https://www.codeshelter.co/

## TL;DR

I'm going to be away for 2 years and unable to maintain Squirrelly.

In case I am unable to find a maintainer before I leave, I will give repository access to a trusted friend and add the repository to https://www.codeshelter.co/ (allowing anyone to apply for maintainership).

If you want to become one of Squirrelly's maintainers and have experience with open-source and TypeScript, message me on [Gitter](https://gitter.im/nebrelbug), email me at nebrelbug [at] gmail [dot] com, or contact me some other way!

**Comment by nebrelbug** [comment]
If anybody would like to contribute to Squirrelly, I've added it to [Code Shelter](https://www.codeshelter.co/). They'll be able to vet potential contributors and grant edit and release access :)

**Comment by kerosin** [comment]
Is Eta also affected?

Thanks anyway for your great work ❤️

**Comment by iamfj** [comment]
What about an union of a few developers to take over the maintenance of squirrelly?

**Comment by cdtut** [comment]
@nebrelbug When someone should use this instead of Eta? They do almost same thing I can't tell what is the difference.

**Comment by nebrelbug** [comment]
@cdtut Eta is actively maintained and targeted towards minimalist embedded JS templating. Squirrelly is for more complex templating with helpers, filters etc.
